app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.java

```diff
@@ -138,6 +145,17 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         searchView =  (TextView) view.findViewById(R.id.search_hint);
         searchView.setOnClickListener(this);
 
+        searchViewClip =  (TextView) view.findViewById(R.id.search_hint_clip);
+        searchViewClip.setOnClickListener(this);
+
+        ClipboardManager clipboard = (ClipboardManager) getContext().getSystemService(CLIPBOARD_SERVICE);
+        ClipData clip = clipboard.getPrimaryClip();
+
+        if (clip != null && clip.getItemCount() > 0) {
+            ClipData.Item item = clip.getItemAt(0);
+            pasteString = item.getText().toString();
```

Comment:
I think `item.getText()` can return `null` if paste doesn't contain text (not sure however)?